File: project/frontend/frontend/src/app/services/ticket.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import { Ticket } from "../modules/models/ticket";


@Injectable()
// Data service
export class TicketService { //todo create interface

  constructor(private http: HttpClient) {
  }

  // Ajax request for billing account data
  getTickets(): Observable<Ticket[]> {
    return this.http.get<Ticket[]>('/api/ticket')
  }

  saveTicket(ticket: Ticket): Observable<Ticket> {
    return this.http.post<Ticket>('/api/ticket', ticket);
  }

  deleteTicket(idTicket: number): Observable<void> {
    return this.http.delete<void>('/api/ticket/' + idTicket);
  }
  
}

File: project/fapi/src/main/java/com/netcracer/edu/fapi/service/TicketService.java
package com.netcracer.edu.fapi.service;

import com.netcracer.edu.fapi.models.Ticket;

import java.util.List;

public interface TicketService {
    List<Ticket> findAll();
    Ticket save(Ticket ticket);
    void delete(Integer idTicket);
}
